catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DumpCommand.java

```diff
@@ -239,9 +248,8 @@ protected Object executeWithSubject() throws Exception {
                 break;
             }
 
-            if (pageSize > 0) {
-                query.setStartIndex(query.getStartIndex() + pageSize);
-            }
+            query.setStartIndex(query.getStartIndex() + pageSize);
```

Comment:
❓ So...you're doing your own paging on top of the paging being run by the `ResultIterable`. Might it make more sense to just partition the results from the iterable and let it handle the pagination internally?